src/components/ImageView/index.tsx

```diff
@@ -1,41 +1,81 @@
+import purify from 'dompurify';
 import log from 'loglevel';
 import * as React from 'react';
 
+import { gettext } from '../../utils';
 import styles from './styles.module.scss';
 
-type PublicProps = {
+export type PublicProps = {
   _btoa?: typeof btoa;
   _log?: typeof log;
+  _sanitize?: typeof purify.sanitize;
   mimeType: string;
   content: string;
 };
 
+// From https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#Firefox.
+const supportedImageMimeTypes = [
+  'image/jpeg',
+  'image/gif',
+  'image/png',
+  'image/svg+xml',
+  'image/bmp',
+  'image/vnd.microsoft.icon',
+  // webp isn't listed in the MDN article, but is documented elsewhere.
+  // e.g., https://caniuse.com/#search=webp
+  'image/webp',
+];
+
 type ConvertImageDataParams = {
   _btoa?: typeof btoa;
   _log?: typeof log;
+  _sanitize?: typeof purify.sanitize;
   content: string;
+  mimeType: string;
 };
 
-export const convertImageData = ({
+const convertImageData = ({
   _btoa = btoa,
   _log = log,
+  _sanitize = purify.sanitize,
   content,
+  mimeType,
 }: ConvertImageDataParams): string | null => {
-  try {
-    return _btoa(content);
-  } catch (error) {
-    _log.debug('Error converting image data to ascii:', error);
+  const mimeTypeCased = mimeType.toLowerCase();
+  if (!supportedImageMimeTypes.includes(mimeTypeCased)) {
+    _log.error('Unrecognized mimeType :', mimeType);
     return null;
   }
+
+  return _btoa(
+    // Run sanitize on svg files.
+    mimeTypeCased === 'image/svg+xml' ? _sanitize(content) : content,
+  );
 };
 
-const ImageViewBase = ({ _btoa, _log, content, mimeType }: PublicProps) => {
-  const imageData = convertImageData({ _btoa, _log, content });
-  return imageData ? (
+const ImageViewBase = ({
+  _btoa,
+  _log,
+  _sanitize,
+  content,
+  mimeType,
+}: PublicProps) => {
+  const imageData = convertImageData({
+    _btoa,
```

Comment:
TypeScript makes it safe enough to use spreads here:

```diff
diff --git a/src/components/ImageView/index.tsx b/src/components/ImageView/index.tsx
index 26934ef..8e6955a 100644
--- a/src/components/ImageView/index.tsx
+++ b/src/components/ImageView/index.tsx
@@ -53,20 +53,8 @@ const convertImageData = ({
   );
 };
 
-const ImageViewBase = ({
-  _btoa,
-  _log,
-  _sanitize,
-  content,
-  mimeType,
-}: PublicProps) => {
-  const imageData = convertImageData({
-    _btoa,
-    _log,
-    _sanitize,
-    content,
-    mimeType,
-  });
+const ImageViewBase = ({ mimeType, ...props }: PublicProps) => {
+  const imageData = convertImageData({ mimeType, ...props });
   return (
     <div className={styles.ImageView}>
       {imageData ? (
```

Without TypeScript it's risky because you could have a hard time keeping track of the spread properties.